xarray/tests/test_coding_strings.py

```diff
@@ -92,6 +93,14 @@ def test_CharacterArrayCoder_roundtrip(original):
     coder = strings.CharacterArrayCoder()
     roundtripped = coder.decode(coder.encode(original))
     assert_identical(original, roundtripped)
+    if original.encoding == {}:
```

Comment:
This is a little funky, but fairly clear. If there is no original char_dim_name, the decoded version has the default "stringN", otherwise the char_dim_names match.

Reply:
We try to avoid writing any logic in tests because it makes things harder to follow. Would you mind writing a separate test function for just this case?

It would also be good to explicitly test the name of the new dimension on the encoded variable.